src/rpc/dial.rs

```diff
@@ -470,6 +464,20 @@ impl<T: AuthMethod> DialBuilder<T> {
 }
 
 impl DialBuilder<WithoutCredentials> {
+    fn clone(&self) -> Self {
+        DialBuilder {
+            state: WithoutCredentials(()),
+            config: DialOptions {
+                credentials: None,
+                webrtc_options: self.config.webrtc_options.clone(),
+                uri: self.duplicate_uri(),
+                disable_mdns: self.config.disable_mdns,
+                allow_downgrade: self.config.allow_downgrade,
+                insecure: self.config.insecure,
+            },
+        }
+    }
```

Comment:
the `connect()` method consumes a `DialBuilder`, so we need to be able to clone them in order to allow parallel attempts. Incidentally, this allows us to clean up a bit some of the dial logic below.